### PR TITLE
TEST:Add unit test for test delete activity type

### PR DESCRIPTION
**Pull request by Zmxtz**
# Description

Added unit test for test_ delete_activity_type

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if appropriate
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have created or extended unit tests to address my new additions
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

If you have any questions, please contact @macite or @jakerenzella.

### CODE REVIEW

File: test/api/activity_types_api_test.rb [hunk 1]
```diff
@@ -18,8 +18,8 @@ def test_get_all_activity_types
     response_keys = %w(name abbreviation)
 
     last_response_body.each do | data |
-      activity_type = ActivityType.find(data['id'])
-      assert_json_matches_model(data, activity_type, response_keys)
+      expected_data = ActivityType.find(data['id'])
+      assert_json_matches_model(data, expected_data, response_keys)
```
Comment (linhhtr) [inline test/api/activity_types_api_test.rb]: This modification does nothing...

Comment (linhhtr) [inline test/api/activity_types_api_test.rb]: This changes do nothing... why you bother changing the variable name?

File: test/api/activity_types_api_test.rb [hunk 2]
```diff
@@ -50,4 +50,42 @@ def test_put_activity_types
     first_activity_type = ActivityType.first
     assert_json_matches_model(last_response_body, first_activity_type, response_keys)
   end
+
+  def test_delete_activity_type
+    # Create a activity type
+    activity_type = FactoryBot.create(:activity_type)
+    id_activity_type= activity_type.id
```
Comment (linhhtr) [inline test/api/activity_types_api_test.rb]: This variable is redundant as you already use `activity_type.id` in line 62

File: test/api/activity_types_api_test.rb [hunk 3]
```diff
@@ -50,4 +50,42 @@ def test_put_activity_types
...
+        
+     # create a activity type to delete
+    activity_type = FactoryBot.create(:activity_type)
+
+    # perform the delete
+    delete_json with_auth_token("/api/activity_types/#{activity_type.id}", user)
+    
+    # check if the delete does not get through
+    assert_equal 403, last_response.status
+
```
Comment (linhhtr) [inline test/api/activity_types_api_test.rb]: Please add another check to see you still can find the activity type you attempted to delete and another to see the number of activity type is still the same

**Comment by macite** [comment]
Is this ready for review now? Tag me when ready

File: test/api/activity_types_api_test.rb [hunk 4]
```diff
@@ -50,4 +50,42 @@ def test_put_activity_types
...
+    # Check delete if success
+    assert_equal ActivityType.count, number_of_ativity_type-1
+
+    # Check that you can't find the deleted id
+    refute ActivityType.exists?(id_activity_type)
+  end
+
+
+  def test_delete_activity_type_not_auth
```
Comment (linhhtr) [inline test/api/activity_types_api_test.rb]: Please rename this into `test_student_cannot_delete_activity_type`

File: test/api/activity_types_api_test.rb [hunk 5]
```diff
@@ -50,4 +50,55 @@ def test_put_activity_types
     first_activity_type = ActivityType.first
     assert_json_matches_model(last_response_body, first_activity_type, response_keys)
   end
+
+  def test_delete_activity_type
+    # Create a activity type
+    activity_type = FactoryBot.create(:activity_type)
+    # id_activity_type= activity_type.id
```
Comment (linhhtr) [inline test/api/activity_types_api_test.rb]: Please delete this line if u do not use it, do not leave it as a comment

File: test/api/activity_types_api_test.rb [hunk 6]
```diff
@@ -50,4 +50,55 @@ def test_put_activity_types
...
+
+  def test_student_cannot_delete_activity_type
+
+
+    # A user with student role which does not have permision to delete a activity type
+    user = FactoryBot.build(:user, :student)
+    
+    # create a activity type to delete
+    activity_type = FactoryBot.create (:activity_type)
+    id_of_tp = activity_type.id
```
Comment (linhhtr) [inline test/api/activity_types_api_test.rb]: Also this line isn't used... Please delete..

File: test/api/activity_types_api_test.rb [hunk 7]
```diff
@@ -18,8 +18,8 @@ def test_get_all_activity_types
     response_keys = %w(name abbreviation)
 
     last_response_body.each do | data |
-      activity_type = ActivityType.find(data['id'])
-      assert_json_matches_model(data, activity_type, response_keys)
+    expected_data = ActivityType.find(data['id'])
```
Comment (macite) [inline test/api/activity_types_api_test.rb]: indent - it is within the each block above.

Comment (macite) [reply test/api/activity_types_api_test.rb]: Same with line below.

File: test/api/activity_types_api_test.rb [hunk 8]
```diff
@@ -50,4 +50,48 @@ def test_put_activity_types
...
+    # check if the delete does not get through
+    assert_equal 403, last_response.status
+
+    # check if the number of ativity_type is still the same
+    assert_equal ActivityType.count, number_of_ativity_type
+
+    # # Check that you still can find the deleted id
+   assert ActivityType.exists?(activity_type.id)
+
+    end
```
Comment (macite) [inline test/api/activity_types_api_test.rb]: Indentation here... and below.

**Pull request merged**